### PR TITLE
api: Add `gpu_cdi` extension

### PR DESCRIPTION
**Pull request by gabrielmougard**
Add an API extension for adding the CDI mode for GPU passthrough (see https://github.com/canonical/lxd/pull/13562)

### CODE REVIEW

File: doc/api-extensions.md [hunk 1]
```diff
@@ -2515,3 +2515,8 @@ contains the total available logical CPUs available when LXD started.
 ## `vm_limits_cpu_pin_strategy`
 
 Adds a new {config:option}`instance-resource-limits:limits.cpu.pin_strategy` configuration option for virtual machines. This option controls the CPU pinning strategy. When set to `none`, CPU auto pinning is disabled. When set to `auto`, CPU auto pinning is enabled.
+
+## `gpu_cdi_mode`
+
+This adds support for using Container Device Interface (CDI) specification to configure GPU passthrough in LXD containers. The `id` field in GPU devices now accepts CDI identifiers (for example, `{VENDOR_DOMAIN_NAME}/gpu=gpu{INDEX}`) in addition to DRM card IDs. This enables GPU passthrough for devices that don't use PCI addressing (like NVIDIA Tegra iGPUs) and provides a more flexible way to identify and configure GPU devices.
```
Comment (minaelee) [inline doc/api-extensions.md:2521]:
```suggestion
Adds support for using the Container Device Interface (CDI) specification to configure GPU passthrough in LXD containers. The `id` field of GPU devices now accepts CDI identifiers (for example, `{VENDOR_DOMAIN_NAME}/gpu=gpu{INDEX}`) for containers, in addition to DRM card IDs. This enables GPU passthrough for devices that don't use PCI addressing (like NVIDIA Tegra iGPUs) and provides a more flexible way to identify and configure GPU devices.
```

File: doc/api-extensions.md [hunk 2]
```diff
@@ -2515,3 +2515,8 @@ contains the total available logical CPUs available when LXD started.
 ## `vm_limits_cpu_pin_strategy`
 
 Adds a new {config:option}`instance-resource-limits:limits.cpu.pin_strategy` configuration option for virtual machines. This option controls the CPU pinning strategy. When set to `none`, CPU auto pinning is disabled. When set to `auto`, CPU auto pinning is enabled.
+
+## `gpu_cdi_mode`
```
Comment (tomponline) [inline doc/api-extensions.md:2519]: "Mode" seems redundent here. How about just gpu_cdi or gpu_cdi_container¿

Comment (gabrielmougard) [reply doc/api-extensions.md]: OK for `gpu_cdi`. fixed

**Review by tomponline — APPROVED** [review]
Cheers

**Comment by tomponline** [comment]
@gabrielmougard docs failing https://github.com/canonical/lxd/actions/runs/11951051486/job/33313863650?pr=14501

**Review by tomponline — APPROVED** [review]

**Pull request merged**